### PR TITLE
Fixes precision for small number

### PR DESCRIPTION
**Pull request by mukulljangid**
Can go up to 20 digits after decimal point.

## High Level Overview of Change
By default, fractional part will now have 6 digits. And for number with a lot of zeros, the fractional part can increase up to 20.

### Context of Change
The previous implementation restricted the numbers to only 4 digits in the fractional part which didn't let us see a good part of it, sometimes only zeros. If there are more than 20 zeros in the most significant fractional part, the program shall bring it down to 0 with 6 digits in the fractional part.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After
Before:
image
After:
image

### CODE REVIEW

File: src/containers/shared/utils.js [hunk 1]
```diff
@@ -151,7 +151,10 @@ export const getLocalizedCurrencySymbol = (lang = 'en-US', currency = 'USD') =>
   return formatted.split('1')[0].trim();
 };
 
-export const formatLargeNumber = (d = 0, digits = 4) => {
+export const formatLargeNumber = (d = 0, digits = 6) => {
```
Comment (mvadari) [inline src/containers/shared/utils.js:154]: This method is used in a lot of different places across the explorer, are you sure those places aren't affected by your changes?

File: src/containers/shared/utils.js [hunk 2]
```diff
@@ -151,7 +151,10 @@ export const getLocalizedCurrencySymbol = (lang = 'en-US', currency = 'USD') =>
...
+  let digitsLocal = 6;
+  let zeros = 0;
+  let copy = d;
```
Comment (mvadari) [inline src/containers/shared/utils.js:157]: This variable should be named more descriptively

File: src/containers/shared/utils.js [hunk 3]
```diff
@@ -151,7 +151,10 @@ export const getLocalizedCurrencySymbol = (lang = 'en-US', currency = 'USD') =>
   return formatted.split('1')[0].trim();
 };
 
-export const formatLargeNumber = (d = 0, digits = 4) => {
+export const formatLargeNumber = (d = 0, digits = 6) => {
+  let digitsLocal = 6;
```
Comment (mvadari) [inline src/containers/shared/utils.js:155]: This variable should be named more descriptively

**Review by mvadari — CHANGES_REQUESTED** [review]

File: src/containers/shared/utils.js (hunk 3 above)
Comment (mukulljangid) [reply src/containers/shared/utils.js]: Done!

File: src/containers/shared/utils.js (hunk 2 above)
Comment (mukulljangid) [reply src/containers/shared/utils.js]: Done!

File: src/containers/shared/utils.js (hunk 1 above)
Comment (mukulljangid) [reply src/containers/shared/utils.js:154]: It's only for the Tokens page and it works perfectly.

Comment (mvadari) [reply src/containers/shared/utils.js:154]: It's also used in the token page header (the per-token page)

File: src/containers/shared/utils.js [hunk 4]
```diff
@@ -151,7 +151,10 @@ export const getLocalizedCurrencySymbol = (lang = 'en-US', currency = 'USD') =>
   return formatted.split('1')[0].trim();
 };
 
-export const formatLargeNumber = (d = 0, digits = 4) => {
+export const formatLargeNumber = (d = 0, digits = 6) => {
+  let variableDigits = 6;
+  let numberOfZeros = 0;
+  let numberCopy = d;
```
Comment (mvadari) [inline src/containers/shared/utils.js:157]: Looks like you don't actually have to copy `d` - it's not modified anywhere else, so you can just use `d`

File: src/containers/shared/utils.js (hunk 1 above)
Comment (mvadari) [inline src/containers/shared/utils.js:154]: Why did the default number of digits change? You can just pass in a different value for the variable and then you have to worry less about differences

Comment (mukulljangid) [reply src/containers/shared/utils.js:154]: Either way is fine!

File: src/containers/shared/utils.js [hunk 5]
```diff
@@ -152,6 +152,9 @@ export const getLocalizedCurrencySymbol = (lang = 'en-US', currency = 'USD') =>
 };
 
 export const formatLargeNumber = (d = 0, digits = 4) => {
+  let variableDigits = 4;
```
Comment (mvadari) [inline src/containers/shared/utils.js:155]: Why does this start at 4?

File: src/containers/shared/utils.js [hunk 6]
```diff
@@ -187,8 +190,21 @@ export const formatLargeNumber = (d = 0, digits = 4) => {
     };
   }
 
+  while (numberCopy < 1 && variableDigits < 20) {
+    numberCopy *= 10;
+    numberOfZeros += 1;
+    variableDigits = numberOfZeros > variableDigits - 1 ? variableDigits + 1 : variableDigits;
+  }
```
Comment (mvadari) [inline src/containers/shared/utils.js:197]: Leave a comment here that explains what this is doing

Comment (mukulljangid) [reply src/containers/shared/utils.js:197]: Counting the number of zeros after the decimal point and displaying two non-zero digits (eg. 0.00000023/45/50) at the end. Also, we can display a maximum of 20 digits.

File: src/containers/shared/utils.js (hunk 5 above)
Comment (mukulljangid) [reply src/containers/shared/utils.js:155]: The can be very very small. So displaying a minimum of 4 digits might give us a sense. If we try to reduce that, most small numbers won't have significant information to display.

File: src/containers/shared/utils.js (hunk 6 above)
Comment (mvadari) [reply src/containers/shared/utils.js:197]: No, leave a comment in the code

File: src/containers/shared/utils.js (hunk 5 above)
Comment (mvadari) [reply src/containers/shared/utils.js:155]: Why does it not start at `digits`?

File: src/containers/shared/utils.js [hunk 7]
```diff
@@ -151,7 +151,12 @@ export const getLocalizedCurrencySymbol = (lang = 'en-US', currency = 'USD') =>
   return formatted.split('1')[0].trim();
 };
 
+// displaying 4 digits will give user a sense of how small the number is.
+// less number of digits might not have siginificant information since tokens have really small value.
```
Comment (mvadari) [inline src/containers/shared/utils.js:155]: This comment isn't needed anymore, now that we changed `variableDigits` from a magic number.

**Review by mvadari — APPROVED** [review]

**Pull request merged**